src/include/cursor_inline.h

```diff
@@ -440,6 +440,39 @@ __wt_cursor_func_init(WT_CURSOR_BTREE *cbt, bool reenter)
     return (0);
 }
 
+/*
+ * __wt_cursor_free_cached_memory --
+ *     If a cached cursor is still holding memory, free it now.
+ */
+static WT_INLINE void
+__wt_cursor_free_cached_memory(WT_CURSOR *cursor)
+{
+    WT_SESSION_IMPL *session;
+
+    if (F_ISSET(cursor, WT_CURSTD_CACHED_WITH_MEM)) {
```

Comment:
I think it would be better to assert this, otherwise someone could call this in an invalid state.

Reply:
Actually, __wt_cursor_free_cached_memory is called unconditionally from the sweep, so it's needed here.

Reply:
That's true, in that case I'd prefer we check it from within the sweep function and assert here, but it's a minor thing so I'm not fussed.